runtime/doc/syntax.txt

```diff
@@ -2393,10 +2393,15 @@ Comments are also highlighted by default.  You can turn this off by using: >
 
 	:let make_no_comments = 1
 
-Microsoft Makefile handles variable expansion and comments differently
-(backslashes are not used for escape).  If you see any wrong highlights
-because of this, you can try this: >
+There are various Make implementations, which add extensions other than the
+POSIX specification and thus are mutually incompatible.  If the filename is
+BSDmakefile or GNUmakefile, the corresponding implementation is automatically
+detected; otherwise it is determined by the file contents.  If you see any
+wrong highlights because of this, you can enforce a flavor by setting one of the
+following: >
 
+	:let make_bsd = 1  " or
+	:let make_gnu = 1  " or
 	:let make_microsoft = 1
```

Comment:
Perhaps it would be better to introduce `make_flavor="bsd|gnu|microsoft"`, like `tex_flavor`, instead of these variables.

Reply:
That seems like a good idea to me!

Reply:
Can we please (for backwards compatibility) at least still use the deprecated variables `make_gnu`, `make_bsd` and `make_microsoft` as fallback?